modules/templates/helper.go

```diff
@@ -294,24 +294,27 @@ func timeEstimateString(timeSec any) string {
 	return util.TimeEstimateString(v)
 }
 
-func queryBuild(a ...any) template.URL {
+// QueryBuild builds a query string from a list of key-value pairs.
+// It omits the nil and empty strings, but it doesn't omit other zero values,
+// because the zero value of number types may have a meaning.
+func QueryBuild(a ...any) template.URL {
 	var s string
 	if len(a)%2 == 1 {
 		if v, ok := a[0].(string); ok {
 			if v == "" || (v[0] != '?' && v[0] != '&') {
-				panic("queryBuild: invalid argument")
+				panic("QueryBuild: invalid argument")
```

Comment:
Do we really need to panic? Is it possible to recover from a panic if necessary?

Reply:
> Do we really need to panic?

Yes



> Is it possible to recover from a panic if necessary?

No

Reply:
From a user's perspective, no one wants to cause the entire system to panic because of a query. I think this is unreasonable.

Reply:
> From a user's perspective, no one wants to cause the entire system to panic because of a query. I think this is unreasonable.

There are a variety of operations in Golang that may cause a panic. For this case, I think it is the developer's responsibility to check and ensure that the query does not cause a panic.

Reply:
> From a user's perspective, no one wants to cause the entire system to panic because of a query. I think this is unreasonable.

No, it won't panic from "user's perspective". It only panics for developer's mistakes.

It is designed carefully and intentionally, I won't change it unless you could show a real case why end users would be affected.